File: zig/src/algorithms/linear-search.zig
const std = @import("std");

pub fn linear_search(array: []const i32, x: i32) bool {
    for (array) |item| {
        if (item == x) {
            return true;
        }
    }

    return false;
}

test "linear search test" {
    const array = [_]i32{ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 1000 };
    const expect = std.testing.expect;

    try expect(linear_search(&array, 1) == true);
    try expect(linear_search(&array, 1000) == true);
    try expect(linear_search(&array, 5000) == false);
}

File: rust/src/algorithms/linear_search.rs
pub fn linear_search(array: &[i32], x: i32) -> bool {
    for &item in array {
        if item == x {
            println!("true");
            return true;
        }
    }

    return false;
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_linear_search_found() {
        let array = [1, 2, 3, 4, 5];
        assert!(linear_search(&array, 3));
    }

    #[test]
    fn test_linear_search_not_found() {
        let array = [1, 2, 3, 4, 5];
        assert!(!linear_search(&array, 6));
    }
}
